### PR TITLE
fix(strictNull): adding strict null check to lambdas

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

Add strict null checks to the lambdas

### CODE REVIEW

File: lambdas/user-list-search-corpus-indexing/tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
   "extends": "tsconfig/lambda.json",
   "compilerOptions": {
     "outDir": "dist",
-    "rootDir": "src"
+    "rootDir": "src",
+    "strictNullChecks": false
```
Comment (bassrock) [inline lambdas/user-list-search-corpus-indexing/tsconfig.json:6]: Turned off to not conflict with https://github.com/Pocket/pocket-monorepo/pull/751

File: lambdas/account-data-deleter-batch-delete/src/config.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
 const shortName = 'ADD';
 
 const awsEnvironments = ['production', 'development'];
-let localAwsEndpoint = undefined;
-if (!awsEnvironments.includes(process.env.NODE_ENV)) {
+let localAwsEndpoint: string | undefined = undefined;
```
Comment (kschelonka) [inline lambdas/account-data-deleter-batch-delete/src/config.ts:4]: don't need to change this, but I'd prefer a ternary statement vs. initialization and conditional assignment for our code generally

File: lambdas/transactional-emails/src/schemas/premiumPurchaseSchema/premiumPurchaseEvent.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { User } from './user';
 
 export class PremiumPurchaseEvent {
   'purchase': Purchase;
-  'user': User;
+  'user': User | null;
```
Comment (kschelonka) [inline lambdas/transactional-emails/src/schemas/premiumPurchaseSchema/premiumPurchaseEvent.ts:6]: how is this nullable?

File: lambdas/transactional-emails/src/schemas/premiumPurchaseSchema/purchase.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ export class Purchase {
   'isTrial': boolean;
   'planInterval': string;
   'planType': string;
-  'receiptId': string;
+  'receiptId': string | null;
```
Comment (kschelonka) [inline lambdas/transactional-emails/src/schemas/premiumPurchaseSchema/purchase.ts:8]: how is this nullable?

File: lambdas/account-data-deleter-batch-delete/src/config.ts (hunk 2 above)
Comment (bassrock) [reply lambdas/account-data-deleter-batch-delete/src/config.ts:4]: @kschelonka `const localAwsEndpoint =
  process.env.NODE_ENV && !awsEnvironments.includes(process.env.NODE_ENV)
    ? process.env.AWS_ENDPOINT || 'http://localhost:4566'
    : undefined;`

File: lambdas/user-list-search-corpus-indexing/tsconfig.json (hunk 1 above)
Comment (kschelonka) [reply lambdas/user-list-search-corpus-indexing/tsconfig.json:6]: #751 is merged

File: lambdas/user-list-search-corpus-parser-hydration/src/index.ts [hunk 5]
```diff
@@ -121,7 +122,11 @@ export function unwrapPayloads(payload: EventPayload): BulkRequestMeta[] {
       messageId: payload.messageId,
     };
     return [parent, ...stories];
-  } else {
+  } else if (
```
Comment (kschelonka) [inline lambdas/user-list-search-corpus-parser-hydration/src/index.ts:125]: noting that this is already evaluated before passed to the function; will update in follow-up

**Review by kschelonka — APPROVED** [review]

**Pull request merged**